src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets

```diff
@@ -1307,4 +1307,14 @@ Copyright (c) .NET Foundation. All rights reserved.
     ============================================================
   -->
   <Import Project="NuGet.RestoreEx.targets" Condition="'$(RestoreUseStaticGraphEvaluation)' == 'true' And Exists('NuGet.RestoreEx.targets')" />
+
+  <!--
+    ============================================================
+    Displays the preview information message for the CentralPackageVersionManagement
+    ============================================================
+  -->
+  <Target Name="CentralPackageVersionManagementPreview" BeforeTargets="Restore" AfterTargets="AfterBuild" Condition="'$(_CentralPackageVersionsEnabled)'=='true'">
```

Comment:
I'm not sure this the right way to approach this.

In VS, we don't run the restore target, but we have our custom implementation. 
It would happen after build & by then you've already lost the restore context.

Reply:
Updated the logic.